internal/juju/offers.go

```diff
@@ -310,7 +310,16 @@ func (c offersClient) ConsumeRemoteOffer(input *ConsumeRemoteOfferInput) (*Consu
 
 	localName, err := client.Consume(consumeArgs)
 	if err != nil {
-		return nil, err
+		// Check if SAAS is already created. If so return offer response instead of error
+		if strings.Contains(err.Error(), "saas application already exists") {
+			// https://github.com/juju/juju/blob/3e561add5940a510f785c83076b2bcc6994db103/api/client/application/client.go#L803
```

Comment:
please add descriptive test on why a blob url is included.

Reply:
Done